### DISCUSSION TITLE
blog/2024-07-31_zero-cross_pulse_detection

### DISCUSSION THREAD

**Discussion by giscus** [mathieucarbou/YaSolR#31] [Website comments]
# blog/2024-07-31_zero-cross_pulse_detection

Zero-Cross Pulse Detection

https://yasolr.carbou.me/blog/2024-07-31_zero-cross_pulse_detection

**Comment by manumdk** [comment]
Salut,
J'ai utilisé ta bibliothéque pour tester mon routeur DIY, j'ai comme retour :
12:16:03.408 > {"state":0,"period":10002,"period_min":9961,"period_max":10045,"frequency":99.98000336,"length":436,"length_min":428,"length_max":444}

Si le signal avait été mauvais, j'aurai eu une fréquence supérieure ?
Merci pour ton partage.
Manu

**Reply by mathieucarbou** [reply]
L'analyse de pulse en fait dure environ une seconde et s'arrête, elle n'est tpas continue. Et elle sert à déterminer les stats du pulse en évitant les artefacts. Donc ça donne une idée des pulses normaux et de leur largeur, et ça sert à mieux déclencher le triac au bon moment.
